### PR TITLE
all rates

### PR DESCRIPTION
**Pull request by wakamex**
not ready for merge: wanna add continue from existing data

btc
image

eth
image

### CODE REVIEW

File: trueblocks_queries/comp_rates.sh [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
+result_lines_btc=$(wc -l resultsBTC.txt | cut -f1 -d ' ')
+
+# concatenate the results
+tail -n $result_lines_eth blocksETH.txt >blocksETHwithresults.txt
+tail -n $result_lines_btc blocksBTC.txt >blocksBTCwithresults.txt
+paste -d ',' blocksETHwithresults.txt resultsETH.txt >combinedETH.txt
+paste -d ',' blocksBTCwithresults.txt resultsBTC.txt >combinedBTC.txt
+
+# put on power glove and display in terminal
+jp <combinedETH.txt --input csv -xy '[*][0,1]'
```
Comment (villanuevawill) [inline trueblocks_queries/comp_rates.sh]: Please include sample graphs with PR.

File: trueblocks_queries/comp_rates.sh [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+
+# count the number of result lines
+# this is less than the number of blocks
+# because chifra ignores "unripe" blocks
+# more than 28 blocks behind the head
+result_lines_eth=$(wc -l resultsETH.txt | cut -f1 -d ' ')
+result_lines_btc=$(wc -l resultsBTC.txt | cut -f1 -d ' ')
+
+# concatenate the results
+tail -n $result_lines_eth blocksETH.txt >blocksETHwithresults.txt
```
Comment (villanuevawill) [inline trueblocks_queries/comp_rates.sh]: You just want to index the dataset by number of appearance?

File: trueblocks_queries/comp_rates.sh [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
 
-< month_results.txt jp -input csv -xy '[*][0,1]'
+# export all rates
+chifra state --call "$eth | supplyRatePerBlock()(uint)" --file blocksETH.txt --no_header | awk '$5 ~ /^[1-9]/ { print (($5 / 10^18 * 7200 * 365)) }' >resultsETH.txt
+chifra state --call "$btc | supplyRatePerBlock()(uint)" --file blocksBTC.txt --no_header | awk '$5 ~ /^[1-9]/ { print (($5 / 10^18 * 7200 * 365)) }' >resultsBTC.txt
+
+# count the number of result lines
+# this is less than the number of blocks
+# because chifra ignores "unripe" blocks
+# more than 28 blocks behind the head
```
Comment (villanuevawill) [inline trueblocks_queries/comp_rates.sh]: This comment is confusing - this makes me think you're clarifying against a race condition of some sort.

**Review by villanuevawill — CHANGES_REQUESTED** [review]

File: trueblocks_queries/comp_rates.sh (hunk 2 above)
Comment (wakamex) [reply trueblocks_queries/comp_rates.sh]: cutting off the most recent blocks, assuming those are where the results are missing from. i actually think the oldest blocks are the ones missing results now.

Comment (villanuevawill) [reply trueblocks_queries/comp_rates.sh]: I don't understand chifra state reads from the blocks.txt file?

Comment (villanuevawill) [reply trueblocks_queries/comp_rates.sh]: It wouldn't make sense they are off

**Comment by wakamex** [comment]
new feature who dis
image

**Comment by wakamex** [comment]
why am i using text files like a database? 🤔

**Comment by villanuevawill** [comment]
> why am i using text files like a database? 🤔

I don't know, but I like it ;p

**Pull request merged**